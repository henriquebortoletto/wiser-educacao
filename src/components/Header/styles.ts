import styled from "styled-components";

export const Title = styled.h1`
	font-size: 40px;
	font-weight: 400;
	line-height: 48px;
	color: #383e71;
	margin-bottom: 16px;
	@media (max-width: 650px) {
		font-size: 26px;
		line-height: 32px;
		text-align: center;
	}
`;
export const Description = styled.h2`
	font-size: 16px;
	font-weight: 600;
	line-height: 20px;
	color: #989fdb;
	margin-bottom: 60px;
	@media (max-width: 650px) {
		font-size: 12px;
		line-height: 20px;
		text-align: center;
	}
`;
