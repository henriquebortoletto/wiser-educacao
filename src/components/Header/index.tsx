import * as S from "./styles";

interface HeaderProps {
	title: string;
	description: string;
}

const Header: React.FC<HeaderProps> = function ({ title, description }) {
	return (
		<>
			<S.Title>{title}</S.Title>
			<S.Description>{description}</S.Description>
		</>
	);
};

export default Header;
