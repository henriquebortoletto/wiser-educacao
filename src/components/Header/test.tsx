import { render, screen } from "@testing-library/react";

import Header from "./index";

describe("<Header />", () => {
	it("should render the title", () => {
		const { getByText, container } = render(
			<Header
				title="Olá, seja bem vindo!"
				description="Para acessar a plataforma, faça seu login."
			/>
		);
		expect(getByText("Olá, seja bem vindo!")).toBeVisible();
		expect(
			getByText("Para acessar a plataforma, faça seu login.")
		).toBeVisible();
		expect(container.firstChild).toMatchSnapshot();
	});
});
