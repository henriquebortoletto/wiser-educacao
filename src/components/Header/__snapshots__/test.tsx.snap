// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the title 1`] = `
<h1
  class="styles__Title-u7lf7e-0 kqbgSV"
>
  Ol√°, seja bem vindo!
</h1>
`;
